apps/sms/js/thread_ui.js

```diff
@@ -1113,17 +1083,32 @@ var ThreadUI = {
    * @returns {Promise} Promise that is resolved once navigation is completed.
    */
   navigateToComposer: function(parameters) {
+    var draftDiscardPromise;
+
+    // TODO: Here we should check if draft is not saved instead
     if (Compose.isEmpty()) {
-      return Navigation.toPanel('composer', parameters);
-    }
+      draftDiscardPromise = Promise.resolve();
+    } else {
+      draftDiscardPromise = Utils.confirm(
+        'unsent-message-text',
+        'unsent-message-title',
+        { text: 'unsent-message-option-discard', className: 'danger' }
+      ).then(() => this.discardDraft());
+    }
+
+    return draftDiscardPromise.then(() => {
```

Comment:
The logic with thread lookup and draft discard dialog here and in ActivityHandler becomes very similar - so I'd say this patch is intermediate step to unify them somehow..... Just thought